lightning/src/ln/onion_utils.rs

```diff
@@ -1713,6 +1713,86 @@ impl From<u16> for LocalHTLCFailureReason {
 	}
 }
 
+impl fmt::Display for LocalHTLCFailureReason {
```

Comment:
I'm a bit hesitant on adding this, mainly because I'm not sure it adds much since the enum variant names are already nicely descriptive and can be easily cross-referenced to the LDK docs if more detail is desired. I also think one of the reasons we had strings to begin with was due to the lack of a nice enum, which we now have. We'll also need to remember to update these strings if any docs on the enum change, potentially. Don't feel tooo strongly about it though.

Reply:
Wasn't the idea to extend this enum with fields eventually? Currently there is just `Unknown` with a code. In that case, we'd probably re-introduce the formatter then?

Reply:
What about the in-between of reducing this to just print the variant's name unless there's associated data? Being able to de-dup with the variant docs is very appealing to me. 

```
match self {
   Self::UnknownFailureCode { code } => write!(f, "Unknown failure code: {}", code),
  _ => write!(f, "{:?}", self),
}
```

Also has the benefit of not swallowing the unknown error code which we may want to see for debugging.

Reply:
Maybe style it `UnknownFailureCode` in line with the rest

Reply:
[Debug](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=885989434f54b5df056e671bf4538469) actually does this for us so going to just drop `Display`, it's only a few LOC to re-add in future.

Reply:
Hm indeed. Less is more.